### ISSUE TITLE
Pick allows any computed properties of type number

### ISSUE THREAD

**Issue by AlexGalays** [Microsoft/TypeScript#13906]
**TypeScript Version:**  2.1.5 and 2.2.0-dev.20170206

If we take the code meant to solve React's setState typings from here  https://github.com/Microsoft/TypeScript/issues/12793

**Code**

```ts
function setState<T, K extends keyof T>(obj: T, state: Pick<T, K>) {
    for (let k in state) {
        obj[k] = state[k]
    }
}

interface Foo {
  a: string
  b: number
}

let foo: Foo = { a: "hello", b: 42 }

const someIndex = 8000
setState(foo, { [someIndex]: 'whatever' })
```

**Expected behavior:**
This should not compile. 8000 is not part of the keys of Foo

**Comment by AlexGalays** [comment]
Any chance this could make it for 2.3?

**Comment by RyanCavanaugh** [comment]
@ahejlsberg bug or something else going on here?

**Comment by AlexGalays** [comment]
Good news, this is now fixed in TS 2.6.1

![image](https://user-images.githubusercontent.com/1375586/32552474-5df254e6-c494-11e7-9064-d675bddd4f4d.png)
